### ISSUE TITLE
Merge video files, in which one of them has no audio track

### ISSUE THREAD

**Issue by Zer0bee** [sannies/mp4parser#296]
I want to merge some video files, in which any one of them may not have audio track. Currently, if i merge these files, audio is moving into another segment, which is wrong. This is how, i am merging files

        LinkedList<Track> videoTracks = new LinkedList<>();
        LinkedList<Track> audioTracks = new LinkedList<>();

        for (Movie m : inMovies) {
            for (Track t : m.getTracks()) {
                if (t.getHandler().equals("soun")) {
                    audioTracks.add(t);
                }
                if (t.getHandler().equals("vide")) {
                    videoTracks.add(t);
                }
            }
      }

        Movie result = new Movie();
        if (audioTracks.size() > 0) {
            result.addTrack(new AppendTrack(audioTracks.toArray(new Track[audioTracks.size()])));
        }
        if (videoTracks.size() > 0) {
            result.addTrack(new AppendTrack(videoTracks.toArray(new Track[videoTracks.size()])));
        }

Video file that does not have audio, will not go inside `if (t.getHandler().equals("soun"))`

**Comment by CesarJNasr** [comment]
have you found a solution ?

**Comment by Zer0bee** [comment]
Yes, you have to add silent audio track for the segment not having audio.

**Comment by stonyz** [comment]
@Zer0bee how to add silent audio, can you give sample code?
